File: src/Library/Client/RcpTransactedCommandAttribute.vb
Imports System.Threading

''' <summary>
''' Contains metadata about Transacted RCP commands.
''' </summary>
''' <remarks></remarks>
<AttributeUsage(AttributeTargets.Method, allowmultiple:=False)> _
Public NotInheritable Class RcpTransactedCommandAttribute
    Inherits Pixa.Soundbridge.Library.RcpCommandAttribute

    Public Sub New(ByVal command As String)
        MyBase.New(command)
    End Sub

    Public Overrides Function CreateResponseProcessor(ByVal client As TcpSoundbridgeClient, ByVal waitHandle As System.Threading.EventWaitHandle) As IResponseProcessor
        Return New TransactedResponseProcessor(client, Command, waitHandle)
    End Function
End Class

File: src/Pixa.Soundbridge/Client/RcpTransactedCommandAttribute.cs
using System;
using System.Threading;

namespace Pixa.Soundbridge.Client {

    /// <summary>
    /// Contains metadata about Transacted RCP commands.
    /// </summary>
    /// <remarks></remarks>
    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
    public sealed class RcpTransactedCommandAttribute : RcpCommandAttribute {
        public RcpTransactedCommandAttribute(string command) : base(command) {
        }

        // TODO: Implement IsList
        public RcpTransactedCommandAttribute(string command, bool isList) : this(command) {
        }

        public override IResponseProcessor CreateResponseProcessor(TcpSoundbridgeClient client, EventWaitHandle waitHandle) {
            return new TransactedResponseProcessor(client, Command, waitHandle);
        }
    }
}